src/OrleansProviders/Streams/Generator/GeneratorPooledCache.cs

```diff
@@ -99,7 +153,10 @@ private ArraySegment<byte> SerializeMessageIntoPooledSegment(GeneratedBatchConta
                 {
                     // no block or block full, get new block and try again
                     currentBuffer = bufferPool.Allocate();
-                    currentBuffer.SetPurgeAction(PurgeAction);
+                    if (this.OnBlockAllocated == null)
+                        throw new OrleansException("Eviction strategy's OnBlockAllocated is not set for current data adapter, this will affect cache purging");
```

Comment:
I'm not sure we should consider this an error at this time.  Remember that queues can be shut down and moved between silos.  Until we've tested that this new code behaves well in those scenarios, let's not assume that data adapters -must- always be wired up to eviction strategies.  The risk here is only that a block does not return to the pool, which is not good, whereas if we throw it could lead to data loss.  Logging a warning would be better than a throw here, if we're concerned about this possibility.